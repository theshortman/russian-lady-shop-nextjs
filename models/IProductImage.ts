export interface IProductImage {
  id: number;
  image_large: string;
  image_medium: string;
  image_small: string;
}
