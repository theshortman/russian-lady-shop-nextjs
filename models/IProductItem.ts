export interface IProductItem {
  id: number;
  size: number;
  quantity: number;
}
