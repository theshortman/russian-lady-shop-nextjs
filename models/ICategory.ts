import { IProduct } from "@/models/IProduct";

export interface ICategory {
  id: number;
  name: string;
  description: string;
  slug: string;
  products: IProduct[];
  prev_page_number: number | null;
  next_page_number: number | null;
}
