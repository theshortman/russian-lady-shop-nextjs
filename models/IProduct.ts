import { IProductImage } from "@/models/IProductImage";
import { IProductItem } from "@/models/IProductItem";

export interface IProduct {
  id: number;
  name: string;
  slug: string;
  description: string;
  detail: string;
  price: number;
  discount: number;
  new_price: number;
  product_images: IProductImage[];
  product_items: IProductItem[];
}
