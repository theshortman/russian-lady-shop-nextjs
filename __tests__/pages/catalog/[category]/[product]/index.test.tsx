import { screen, fireEvent, render } from "@testing-library/react";
import ProductPage from "@/pages/catalog/[category]/[product]";
import { IProduct } from "@/models/IProduct";

describe("Product page", () => {
  const product: IProduct = {
    id: 1,
    name: "Test name",
    slug: "test-slug",
    description: "Test description",
    detail: "test detail",
    price: 1000,
    discount: 10,
    new_price: 900,
    product_images: [
      {
        id: 1,
        image_large: "first-image-large",
        image_medium: "first-image-medium",
        image_small: "first-image-small",
      },
      {
        id: 2,
        image_large: "second-image-large",
        image_medium: "second-image-medium",
        image_small: "second-image-small",
      },
    ],
    product_items: [
      {
        id: 1,
        size: 50,
        quantity: 1,
      },
    ],
  };

  test("renders product page unchanged", () => {
    const { container } = render(<ProductPage product={product} />);
    expect(container).toMatchSnapshot();
  });

  test("should have src of the main image", () => {
    render(<ProductPage product={product} />);

    const img = screen.getByTestId("product-img");
    expect(img).toHaveAttribute("src", "first-image-large");
  });

  test("should change src of the main image by click", async () => {
    render(<ProductPage product={product} />);
    const thumbnails = await screen.findAllByTestId("thumbnail-img");
    fireEvent.click(thumbnails[1]);

    const img = screen.getByTestId("product-img");
    expect(img).toHaveAttribute("src", "second-image-large");
  });
});
