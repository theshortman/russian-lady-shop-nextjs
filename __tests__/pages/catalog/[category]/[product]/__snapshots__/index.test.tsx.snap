// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page renders product page unchanged 1`] = `
<div>
  <div>
    <h1>
      Test name
    </h1>
    <div
      class="row"
    >
      <div
        class="col-sm-6 col-md-5 mb-2"
      >
        <div>
          <svg
            aria-label="No photo"
            class="img-fluid img-thumbnail"
            focusable="false"
            height="912"
            preserveAspectRatio="xMidYMid slice"
            role="img"
            width="600"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#ffffff"
              height="100%"
              width="100%"
            />
            <text
              dy=".3em"
              fill="#dee2e6"
              text-anchor="middle"
              x="50%"
              y="50%"
            >
              Loading...
            </text>
          </svg>
          <img
            alt="Test name"
            class="img-fluid img-thumbnail d-none"
            data-testid="product-img"
            src="first-image-large"
          />
          <img
            alt="Test name"
            class="img-fluid img-thumbnail mt-1 me-1 d-none"
            data-testid="thumbnail-img"
            src="first-image-small"
            style="cursor: pointer;"
          />
          <img
            alt="Test name"
            class="img-fluid img-thumbnail mt-1 me-1 d-none"
            data-testid="thumbnail-img"
            src="second-image-small"
            style="cursor: pointer;"
          />
        </div>
      </div>
      <div
        class="col-sm-6 col-md-7"
      >
        <div
          class="me-1 badge bg-secondary"
        >
          1000
           RUB
        </div>
        <div
          class="badge bg-danger"
        >
          -
          10
          %
        </div>
        <div
          class="fs-5 fw-bold text-success"
        >
          900
           RUB
        </div>
        <div
          class="mb-3"
        >
          Вы экономите
           
          <span
            class="fw-bold text-danger"
          >
            100
             RUB
          </span>
        </div>
        <div
          class="mb-2"
        >
          <strong>
            Размеры в наличии:
          </strong>
        </div>
        <form
          action=""
          class="mb-3"
          method="post"
        >
          <div
            class="mb-2"
          >
            <label
              class="me-2"
            >
              <input
                class="me-1"
                name="size"
                type="radio"
                value="1"
              />
              50
            </label>
          </div>
          <div
            class="form-group"
          >
            <button
              class="btn btn-danger text-white"
              disabled=""
              type="button"
            >
              ДОБАВИТЬ В КОРЗИНУ
            </button>
          </div>
        </form>
        <p>
          Test description
        </p>
        <div>
          test detail
        </div>
      </div>
    </div>
  </div>
</div>
`;
